### ISSUE TITLE
Secondary Y axis

### ISSUE THREAD

**Issue by AdamTheisen** [ANL-DIGR/ACT#163]
@rcjackson @kenkehoe I was looking at adding in a feature for working with a secondary y axis on the timeseries plots, but it is a little more in depth than I thought, mainly due to the set_yrng function.  We rely on the subplot indices to set a lot of things and that currently does not include anything about a secondary y-axis.  I could put in a work around for now that I think will work, but it's not ideal long-term.  Could we adjust the subplot indices at all or are there any other options you could think of?

**Comment by AdamTheisen** [comment]
FYI... I have a very basic secondary y-axis added in, but I do think we need to come up with a better option long-term

**Comment by AdamTheisen** [comment]
@zssherman @mgrover1  I might need to revisit this and see if it's still an issue.  I can't remember what exactly I was trying to do with the second y-axis at the time but will try some tests.  If you can think of ways we could improve our secondary y-axis handling, I'm all ears!

**Comment by kenkehoe** [comment]
I would like a better interaction with the secondary y axis plots. Would be nice to have the axis labels match the color of plotted line since we don't need to use a legend. I also have inconsistent results. If I can find some time I will take a look at this.

**Comment by zssherman** [comment]
@kenkehoe One way of copying over the color to the labels could be something such as this:

```
import numpy as np
import matplotlib.pyplot as plt
x = np.arange(0, 10, 0.1)
y1 = 0.05 * x**2
y2 = -1 *y1

fig, ax1 = plt.subplots()

ax2 = ax1.twinx()
p = ax1.plot(x, y1, 'g-')
p2 = ax2.plot(x, y2, 'b-')

ax1.set_xlabel('X data')
ax1.set_ylabel('Y1 data', color=p[0].get_color())
ax2.set_ylabel('Y2 data', color=p2[0].get_color())

plt.show()
```

I just need to figure out how to implement that in ACT.

**Comment by kenkehoe** [comment]
@zssherman  good idea. I have implemented something similar but not as automated as this in the past. I like your plan. Let me know if you want some help. Implementing this will be used a lot by the DQ Office.

**Comment by zssherman** [comment]
@kenkehoe Revisiting this now, did you want double y axis match colors, just one y match colors?

**Comment by zssherman** [comment]
Have the code in for one, two might not work for now until secondary y is overhauled

**Comment by zssherman** [comment]
Parameter was added for matching color in #632, but will still need to updated secondary y to be more robust.

**Comment by kenkehoe** [comment]
@AdamTheisen  yes/maybe this would break things. I'm sure we can fix our code for something better.
